iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Identifiable.java

```diff
@@ -75,26 +116,201 @@ default Properties getProperties() {
+    }
+
+    /**
+     * Set integer property value associated to specified key.
+     */
+    default int setIntegerProperty(String key, int value) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    /**
+     * Get double property associated to specified key.
+     */
+    default double getDoubleProperty(String key) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    /**
+     * Get double property associated to specified key, with default value.
+     */
+    default double getDoubleProperty(String key, double defaultValue) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    /**
+     * Get double property associated to specified key.
+     */
+    default OptionalDouble getOptionalDoubleProperty(String key) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    /**
+     * Set double property value associated to specified key.
+     */
+    default double setDoubleProperty(String key, double value) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    /**
+     * Get boolean property associated to specified key.
+     */
+    default boolean getBooleanProperty(String key) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    /**
+     * Get boolean property associated to specified key, with default value.
+     */
+    default boolean getBooleanProperty(String key, boolean defaultValue) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    /**
+     * Get boolean property associated to specified key.
+     */
+    default Optional<Boolean> getOptionalBooleanProperty(String key) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    /**
+     * Set boolean property value associated to specified key.
+     */
+    default boolean setBooleanProperty(String key, boolean value) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
 
     /**
      * Get properties key values.
      */
     Set<String> getPropertyNames();
 
+    /**
+     * Get string properties key values.
+     */
+    default Set<String> getStringPropertyNames() {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    /**
+     * Get integer properties key values.
+     */
+    default Set<String> getIntegerPropertyNames() {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    /**
+     * Get double properties key values.
+     */
+    default Set<String> getDoublePropertyNames() {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    /**
+     * Get boolean properties key values.
+     */
+    default Set<String> getBooleanPropertyNames() {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
```

Comment:
Not sure this is useful (what do you think @mathbagu?)

Reply:
Methods remoded